docs/best_practices/extensions.rst

```diff
@@ -45,3 +133,67 @@ anybody who receives the data also receives the necessary data to interpret it.
 .. note::
     In :pynwb-docs:`PyNWB <>`, the extension is cached automatically. This can be specified explicitly with
     ``io.write(filepath, cache_spec=True)``
+
+
+Use Attributes for small metadata related to a particular data object (Group or Dataset)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Attributes should be used mainly to store small metadata (usually less than 64 Kbytes) that
+is associated with a particular Group or Dataset. Typical uses of attributes are, e.g., to
+define the ``unit`` of measurement of a dataset or to store a short ``description`` of
+a group or dataset. For larger data, datasets should be used instead.
+
+In practice, the main difference is that in PyNWB and MatNWB all Attributes are read into memory when reading the
+NWB file. If you would like to allow users to read a file without reading all of these particular data values, use a
+Dataset.
+
+
+Link data to relevant metadata
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Often metadata relevant to a particular type of data is stored elsewhere, e.g., information
+about the ``Device`` used. To ensure relevant metadata can be uniquely identified, the data
+should include links to the relevant metadata. NWB provides a few key mechanisms for linking:
+
+* Use ``links`` (defined via ``NWBLinkSpec``) to link to a particular dataset or group
+* Use ``DynamicTableRegion`` to link to a set of rows in a ``DynamicTable``
+* Use a ``dataset`` with an object reference data type to store collections of links
+  to other objects, e.g., the following dtype to define a dataset of links to ``TimeSeries``
+
+  .. code-block:: yaml
+        dtype:
+            target_type: TimeSeries
+            reftype: object
+
+
+Best practices for object names
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Names for groups, datasets, attributes, or links should typically:
+
+* **Use lowercase letters only**
+* **Use ``_`` instead of `` `` to separate parts in names**. E.g., use the name
+  ``starting_time`` instead of ``starting time``
+* **Explicit**. E.g., avoid the use of ambiguous abbreviation in names.
+
+
+Best practices for naming ``neurodata_types``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For defining new types via ``neurodata_type_def`` use:
+
+* **Use camelcase:**  notation, i.e., names of types should NOT include spaces,
```

Comment:
Well, since CamelCase is specifically mentioned by name here, might as well mention for object names that they use snake case